File: src/my_project/easy_problems/from151To200/substrings_of_size_three.py
from typing import List 

class Solution:
    def countGoodSubstrings(self, s: str) -> int:
        count = 0
        for i in range(2,len(s)):
            temp_string = s[i-2:i+1]
            temp_list = list()

            for j in temp_string:
                if j not in temp_list:
                    temp_list.append(j)
                
            if len(temp_list) == 3:
                count += 1


        return count



solution = Solution()

print(solution.countGoodSubstrings(s = "xyzzaz"))

print(solution.countGoodSubstrings(s = "aababcabc"))

File: my_project_typescript/easy_problems_typescript/from51To100/substringsOfSizeThree.ts
function countGoodSubstrings(s: string): number {

    let count: number = 0
    let tempString: string = ''
    let tempList: Array<string> = []
    for (let i = 2; i < s.length; i++){
        tempString = s.slice(i-2,i+1)
        // console.log(tempString)
        tempList = []
        for (let j of tempString){
            if (!tempList.includes(j)){
                tempList.push(j)
            }
        }
        if (tempList.length === 3){
            count += 1
        }
    }
    return count 

};


console.log(countGoodSubstrings("xyzzaz"))

console.log(countGoodSubstrings("aababcabc"))
